api/ble/README.md

```diff
@@ -0,0 +1,305 @@
+    - [error](#oncherror)
+  * `BluetoothCharacteristic methods
+    - [startNotifications()](#startnotifications)
+    - [startNotifications()](#stopnotifications)
+- [The `BluetoothCharacteristicRequest` interface](#characteristicrequest)
+  * `BluetoothCharacteristicRequest` methods
+    - [respond(data)](#respond)
+    - [error(error)](#errormethod)
+
+Introduction
+------------
+This API exposes Bluetooth Smart functionality for Bluetooth Smart Peripheral (BSP) mode. The terminology and context is taken from the [Bluetooth 4.2. specification](https://www.bluetooth.org/DocMan/handlers/DownloadDoc.ashx?doc_id=286439).
+
+A Bluetooth Smart Peripheral (BSP) device supports the following functionality:
+- Host a [GATT (Generic Attribute Profile)](https://www.bluetooth.com/specifications/gatt) server that exposes a hierarchy of [`Service`](#service), [`Characteristic`](#characteristic) and [`Descriptor`](#descriptor) objects.
+- Broadcast (advertise) information (e.g. services) using advertisment packets. Advertisement or Scan Response data packet is a 31 byte payload used to advertise the device's capabilities and connection parameters.
+- Accept connections from Bluetooth Smart devices in Central mode that can access its services exposed through the GAP, the [Generic Access Profile](https://www.bluetooth.org/en-us/specification/assigned-numbers/generic-access-profile). A device has only one
+instance of the GAP service in the GATT server. The GAP service is a GATT
+based service with the service UUID as «GAP Service» defined in the
+[Bluetooth Assigned Numbers document](https://www.bluetooth.com/specifications/assigned-numbers).
+
+The [W3C Web Bluetooth](https://webbluetoothcg.github.io/web-bluetooth) API exposes functionality for Bluetooth Smart clients (Central mode).
+
+### Use cases
+
+This API is used by applications that implement a Bluetooth Peripheral device and expose sensorial data via Bluetooth.
+- Applications define the Bluetooth Smart services, characteristics and descriptors, then registers them with the implementation.
+- Applications can build Bluetooth advertisement packets, then start (and stop) advertising.
+- For the defined characteristics, applications should register event listeners for handling requests coming from clients. The event listeners receive the request, and should reply to them with appropriate data, or error code.
+
+<a name="apiobject">
+The API object
+--------------
+The API entry point is a [`BluetoothPeripheralDevice`](./#bpd) object that is exposed in a platform specific manner. As an example, on Node.js it can be obtained by requiring the package that implements this API. On other platforms, it can be constructed.
+
+```javascript
+var bpd = require('bluetooth-peripheral');
+```
+If the functionality is not supported by the platform, `require` should throw `NotSupportedError`. If there is no permission for using the functionality, `require` should throw `SecurityError`.
```

Comment:
ok, changed the text to "If the underlying platform doesn't allow using the functionality, `require` should throw `SecurityError`."